src/DateTime.cc

```diff
@@ -496,6 +499,9 @@ struct tm DateTime::gmtime(Timescale scale) const {
 }
 
 struct timespec DateTime::timespec(Timescale scale) const {
+    if (!isValid()) {
+        throw LSST_EXCEPT(pexEx::RuntimeError, "DateTime not valid");
+    }
```

Comment:
My one worry about adding _assertValid is that the exception will show a less useful line number. But I think it's still worth doing (and cleaner than using a macro).